front/pages/api/lookup.ts

```diff
@@ -0,0 +1,192 @@
+import { getSession as getAuth0Session } from "@auth0/nextjs-auth0";
+import type { WithAPIErrorResponse } from "@dust-tt/types";
+import { isLeft } from "fp-ts/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import config from "@app/lib/api/config";
+import { getBearerToken } from "@app/lib/auth";
+import { getPendingMembershipInvitationWithWorkspaceForEmail } from "@app/lib/iam/invitations";
+import type { SessionWithUser } from "@app/lib/iam/provider";
+import { fetchUserFromSession } from "@app/lib/iam/users";
+import { findWorkspaceWithVerifiedDomain } from "@app/lib/iam/workspaces";
+import { Workspace } from "@app/lib/models/workspace";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+export type BaseLookupResponse = {
+  isNew: boolean;
+};
+
+export type UserLookupResponse = BaseLookupResponse & {
+  hasInvite: boolean;
+  hasAutoJoinWorkspace: boolean;
+  workspaceId?: string;
+};
+
+export type WorkspaceLookupResponse = BaseLookupResponse;
+
+const ExternalUserCodec = t.type({
+  email: t.string,
+  email_verified: t.boolean,
+  name: t.string,
+  nickname: t.string,
+  sub: t.string,
+  family_name: t.union([t.string, t.undefined]),
+  given_name: t.union([t.string, t.undefined]),
+  picture: t.union([t.string, t.undefined]),
+});
+
+export type LookupResponseBody = UserLookupResponse | WorkspaceLookupResponse;
+
+const UserLookupSchema = t.type({
+  user: ExternalUserCodec,
+});
+
+const WorkspaceLookupSchema = t.type({
+  workspace: t.string,
+});
+
+const ResourceType = t.union([t.literal("user"), t.literal("workspace")]);
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<LookupResponseBody>>
+): Promise<void> {
+  if (req.method !== "POST") {
+    return apiError(req, res, {
+      status_code: 405,
+      api_error: {
+        type: "method_not_supported_error",
+        message: "Only POST requests are supported",
+      },
+    });
+  }
+
+  const bearerTokenRes = await getBearerToken(req);
+  if (bearerTokenRes.isErr()) {
+    return apiError(req, res, {
+      status_code: 401,
+      api_error: {
+        type: "not_authenticated",
+        message: "The request does not have valid authentication credentials",
+      },
+    });
+  }
+
+  if (bearerTokenRes.value !== config.getLookUpBearerToken()) {
+    return apiError(req, res, {
+      status_code: 401,
+      api_error: {
+        type: "invalid_basic_authorization_error",
+        message: "Invalid token",
+      },
+    });
+  }
+  const { resource } = req.query;
+  if (!resource || typeof resource !== "string") {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Missing or invalid resource parameter",
+      },
+    });
+  }
```

Comment:
Do you need this block ? (i would imagine that `ResourceType.decode(resource);` takes care of it)

Reply:
yeah you right